contracts/governance/BaseWeightedMultisig.sol

```diff
@@ -59,23 +59,27 @@ abstract contract BaseWeightedMultisig is IBaseWeightedMultisig {
      * @param signatures The signatures data
      * @return isLatestSigners True if provided signers are the current ones
      */
-    function validateProof(
-        bytes32 messageHash,
-        bytes calldata weightedSigners,
-        bytes[] calldata signatures
-    ) public view returns (bool isLatestSigners) {
+    function validateProof(bytes32 messageHash, bytes calldata proof) public view returns (bool isLatestSigners) {
         WeightedMultisigStorage storage slot = _baseWeightedStorage();
-        bytes32 signersHash = keccak256(weightedSigners);
+
+        if (proof.length < 32) revert InvalidSigners();
```

Comment:
Is this needed? Can we rely on abi.decode to fail instead?

Reply:
In this case we will get an ambiguous revert message from EVM. I am ok with removing this check if you want. Or we can actually make the number to be 32 * 4.